### DISCUSSION TITLE
Decoding gif failure

### DISCUSSION THREAD

**Discussion by Adonis-Song** [nothings/stb#1634] [Q&A] [answered]
Hello.  I want to use stbi decode gif. These is my code. 
```
STBIDEF stbi_uc *stbi_load_gif(char const *filename, int **delays, int *width, int *height, int *frames, int *nrChannels, int req_comp) 
{
    FILE *f = stbi__fopen(filename, "rb");
    unsigned char *result;
    if (!f) return stbi__errpuc("can't fopen", "Unable to open file");
    fseek(f, 0, SEEK_END);
    int bufSize = ftell(f);
    unsigned char *buf = (unsigned char *)malloc(sizeof(unsigned char) * bufSize);
    if (buf) 
    {
        fseek(f, 0, SEEK_SET);
        fread(buf, 1, bufSize, f);
        result = stbi_load_gif_from_memory(buf, bufSize, delays, width, height, frames, nrChannels, req_comp);
        free(buf);
        buf = NULL;
    }
    else 
    {
        result = stbi__errpuc("outofmem", "Out of memory");
    }
    fclose(f);
    return result;
}
```

I find some gif files could be decoded successfully but some failed.
After debug code,  I find all failed cases are caused by the [this cases](https://github.com/nothings/stb/blob/ae721c50eaf761660b4f90cc590453cdb0c2acd0/stb_image.h#L6743) being triggered during lzw decoding.

I find ffmpeg could decodecd all gif file successfully andIt will not return directly in the case above. 

So is this a stbi bug?

This is a test file which will be decoded failed by stbi
[test.zip](https://github.com/nothings/stb/files/15131146/test.zip)

**Comment by nothings** [comment, marked as answer]
I mean, it could be a bug in stbi, but more likely it's the encoder violating the spec. See for example https://github.com/nothings/stb/issues/1222#issuecomment-938027720

**Comment by Adonis-Song** [comment]
Thank you very much for your reply！

**Comment by ghost** [comment]
https://github.com/nothings/stb/discussions/1634#discussioncomment-9237412

Kind of a hilarious way of saying it with more words. Good entertainment. Where it reads -

```
I don't care, works for me.
```

I won't say it like that. There are better ways to say it in short.

------

To balance it with the other party, the consumer of the free libraries.

If you didn't check it, it's programmer error. But if you checked the magic bytes with stb set of libraries, and then it failed. While other software does not fail. Use whatever works for you.

Song, If your end goal is using a set of stb libraries I won't be able to help otherwise feel free to get in contact.

My name -  Manish
Check out my kind of oldfangled, kinda newfangled, kinda cool, inevitably thought-provoking profile. And the limited mentorship opportunities for your no matter the non business major you have chosen be it - Bachelor's or Master's or Doctorates or Post-Doctorates. 

(yeah I know. I hate signing comments but everyone does shameless self adverts in public, face to face and on here. It is one of many life's bare necessities)

**Reply by ghost** [reply]
Also remind me to complain to Github about issue #345 , subtly incorrect namespaces and networks- NaN.

**Reply by ghost** [reply]
Anyway this made me learn how to block a text of strong opinions that are wrong on Wikipedia. And something about correctly handling the GIF format taht 99.99% of software engineers get wrong.
